s3fs/core.py

```diff
@@ -936,6 +937,14 @@ async def _pipe_file(self, path, data, chunksize=50 * 2 ** 20, **kwargs):
 
     async def _put_file(self, lpath, rpath, chunksize=50 * 2 ** 20, **kwargs):
         bucket, key, _ = self.split_path(rpath)
+
+        # Infer Content Type/Encoding
+        mimetype, encoding = guess_type(key)
+        if isinstance(mimetype, str):
+            kwargs["ContentType"] = kwargs.get("ContentType", mimetype)
```

Comment:
I think it's a bad idea to guess this for the user, where octet-stream is the *right* type for almost any opaque remote blob. Furthermore, the guessing is often wrong :|

I don't actually understand how you can guess the encoding!